### PR TITLE
#0: add suuport for in1 dram sharded matmul2d

### PR DESCRIPTION
**Pull request by yugaoTT**

### CODE REVIEW

File: tt_eager/tt_dnn/op_library/bmm/kernels/dataflow/reader_bmm_tile_layout_in1_sender_writer_padding.cpp [hunk 1]
```diff
@@ -94,6 +94,17 @@ void kernel_main() {
     };
     #endif
 
+    // RT and COMPILE TIME ARGS for DRAM sharded weights
+    #ifdef IN1_DRAM_SHRADED
```
Comment (bbradelTT) [inline tt_eager/tt_dnn/op_library/bmm/kernels/dataflow/reader_bmm_tile_layout_in1_sender_writer_padding.cpp:98]: ...SHARDED?

File: tt_eager/tt_dnn/op_library/bmm/kernels/dataflow/reader_bmm_tile_layout_in1_sender_writer_padding.cpp [hunk 2]
```diff
@@ -213,6 +269,32 @@ void kernel_main() {
...
+                    uint32_t in3_block_w_dram = in1_block_w_dram_stride_bytes[index_offset] / bias_single_tile_size_bytes;
+
+                    for (uint32_t w = 0; w < in3_block_w_dram; ++w) {
+                        noc_async_read_tile_dram_sharded_with_state(in3_base_addr, l1_read_addr_in3, l1_write_addr_in3);
+                        l1_read_addr_in3 += bias_single_tile_size_bytes;
+                        l1_write_addr_in3 += bias_single_tile_size_bytes;
+                        in3_block_size_bytes += bias_single_tile_size_bytes;
+                    }
+                    l1_write_addr_in3_offset += in1_block_w_dram_stride_bytes[index_offset];
+                    index_offset += 2;
```
Comment (bbradelTT) [inline tt_eager/tt_dnn/op_library/bmm/kernels/dataflow/reader_bmm_tile_layout_in1_sender_writer_padding.cpp:294]: Please make the 2 a variable with a descriptive name. There's not enough context to know why you're adding by 2.

Comment (bbradelTT) [reply tt_eager/tt_dnn/op_library/bmm/kernels/dataflow/reader_bmm_tile_layout_in1_sender_writer_padding.cpp:294]: You're also adding by 2 on line 205 above.

Comment (yugaoTT) [reply tt_eager/tt_dnn/op_library/bmm/kernels/dataflow/reader_bmm_tile_layout_in1_sender_writer_padding.cpp]: changed to next_bank_id_and_dram_stride_index

**Review by bbradelTT — APPROVED** [review]

**Pull request merged**